src/Vault.sol

```diff
@@ -542,11 +580,24 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
 
   /**
    * @notice Deposit assets into the Vault and delegate to the sponsorship address.
+   * @dev Will revert if the Vault is under-collateralized.
    * @param _assets Amount of assets to deposit
    * @return uint256 Amount of shares minted to caller.
    */
-  function sponsor(uint256 _assets) external onlyVaultCollateralized returns (uint256) {
-    return _sponsor(_assets, msg.sender);
+  function sponsor(uint256 _assets) external returns (uint256) {
+    address _owner = msg.sender;
+    uint256 _depositedAssets = _totalSupply();
+    uint256 _withdrawableAssets = _totalAssets();
+
+    _depositAssets(_assets, _owner, _owner, _depositedAssets, _withdrawableAssets);
```

Comment:
can inline

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/51/commits/029a830aec2a6519a7c7375dec0ccde99d0e9ef5